src/material-experimental/mdc-slider/slider.ts

```diff
@@ -0,0 +1,517 @@
+  encapsulation: ViewEncapsulation.None,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  providers: [MAT_SLIDER_VALUE_ACCESSOR],
+})
+export class MatSlider implements AfterViewInit, OnChanges, OnDestroy, ControlValueAccessor {
+  /** Event emitted when the slider value has changed. */
+  @Output() readonly change: EventEmitter<MatSliderChange> = new EventEmitter<MatSliderChange>();
+
+  /** Event emitted when the slider thumb moves. */
+  @Output() readonly input: EventEmitter<MatSliderChange> = new EventEmitter<MatSliderChange>();
+
+  /**
+   * Emits when the raw value of the slider changes. This is here primarily
+   * to facilitate the two-way binding for the `value` input.
+   * @docs-private
+   */
+  @Output() readonly valueChange: EventEmitter<number> = new EventEmitter<number>();
+
+  /** Tabindex for the slider. */
+  @Input() tabIndex: number = 0;
+
+  /** The color palette for this slider. */
+  @Input() color: ThemePalette = 'accent';
+
+  /**
+   * Function that will be used to format the value before it is displayed
+   * in the thumb label. Can be used to format very large number in order
+   * for them to fit into the slider thumb.
+   */
+  @Input() displayWith: (value: number) => string | number;
+
+  /** The minimum value that the slider can have. */
+  @Input()
+  get min(): number {
+    return this._min;
+  }
+  set min(value: number) {
+    this._min = coerceNumberProperty(value);
+  }
+  private _min = 0;
+
+  /** The maximum value that the slider can have. */
+  @Input()
+  get max(): number {
+    return this._max;
+  }
+  set max(value: number) {
+    this._max = coerceNumberProperty(value);
+  }
+  private _max = 100;
+
+  /** Value of the slider. */
+  @Input()
+  get value(): number|null {
+    // If the value needs to be read and it is still uninitialized, initialize
+    // it to the current minimum value.
+    if (this._value === null) {
+      this.value = this.min;
+    }
+    return this._value;
+  }
+  set value(value: number|null) {
+    const coercedValue = coerceNumberProperty(value);
+    if (coercedValue !== this._value) {
+      this._value = coercedValue;
+    }
+  }
+  private _value: number|null = null;
+
+  /** The values at which the thumb will snap. */
+  @Input()
+  get step(): number {
+    return this._step;
+  }
+  set step(v: number) {
+    this._step = coerceNumberProperty(v, this._step);
+  }
+  private _step: number = 1;
+
+  /**
+   * How often to show ticks. Relative to the step so that a tick always appears on a step.
+   * Ex: Tick interval of 4 with a step of 3 will draw a tick every 4 steps (every 12 values).
+   */
+  @Input()
+  get tickInterval() {
+    return this._tickInterval;
+  }
+  set tickInterval(value: number|'auto') {
+    if (value === 'auto') {
+      this._tickInterval = 'auto';
+    } else if (typeof value === 'number' || typeof value === 'string') {
+      this._tickInterval = coerceNumberProperty(value, this._tickInterval as number);
```

Comment:
Is this `as number` necessary? `coerceNumberProperty` accepts `any`

Reply:
Yeah shouldn't be necessary. I copied this over from the Material slider. Will fix.